app/models/exam_version.rb

```diff
@@ -379,6 +388,55 @@ def export_files(path, files)
     end
   end
 
+  def walk_files_acc(path, cur_files, and_dirs, &block)
+    cur_files.each do |f|
+      case f['filedir']
+      when 'dir'
+        dpath = path.join(f['path'])
+        yield dpath.to_s if and_dirs
+        walk_files_acc dpath, f['nodes'], and_dirs, &block
+      when 'file'
+        fpath = path.join(f['path'])
+        contents = MarksProcessor.process_marks_reverse(f['contents'], f['marks'])
```

Comment:
Because I'm blanking on context at the moment... we already had called MarksProcessor somewhere, correct?  Is this a _new_ call to it, or replacing/refactoring the existing call?

Reply:
This method is almost exactly `export_files`, but with yields instead of `File.write` and `FileUtils.mkdir_p`